src/engine/Value.ts

```diff
@@ -92,6 +90,44 @@ export abstract class Value<
   }
 }
 
+export class BoolValue extends Value<boolean> {
+  constructor(val: boolean) {
+    super(val || false);
+  }
+  public get isTruthy() {
+    return Boolean(this.value);
+  }
+  public get valueType() {
+    return ValueType.Int;
+  }
+
+  public Cast(newType: ValueType): Value<any> {
```

Comment:
Shouldn't the type be `Value<boolean>`?

Reply:
Mmh no I don't think so, we're casting *from* boolean but to another type. To `newType` but I don't think we can use that in the type declaration. At best we could convert the `any` to `number | text`?

Reply:
`any`'s good your right, I thought we were converting to booleans, sorry!